meson.build

```diff
@@ -160,6 +160,13 @@ if is_firmware_build and get_option('print_memory_usage')
 	add_project_link_arguments('-Wl,--print-memory-usage', language: 'c')
 endif
 
+# We need to know the size of a pointer for the target running the code
```

Comment:
Think you mean "machine" rather than "target" here.

You will need to make a call for checking `sizeof(void *)` for both `cc_host` and `cc_native`, otherwise one of those two is likely going to be wrong for one of either the host or the probe. Use `cc_native` in the `is_cross_build` block.